File: 백준/Silver/7785. 회사에 있는 사람/회사에 있는 사람.py
import sys; input = lambda : sys.stdin.readline().rstrip()

N = int(input())
check = set()

for _ in range(N):
    name, order = tuple(input().split())
    if order == "enter":
        check.add(name)
    else:
        check.remove(name)
print("\n".join(sorted(list(check), reverse=True)))

File: 백준/Silver/7785. 회사에 있는 사람/회사에 있는 사람.java
import java.io.*;
import java.util.*;

public class Main {

    static Set<String> checkMap = new HashSet<>();

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        final int N = Integer.parseInt(br.readLine());
        for(int i = 0; i < N; i++) {
            String[] input = br.readLine().split(" ");
            if (input[1].equals("enter")) {
                checkMap.add(input[0]);
            } else {
                checkMap.remove(input[0]);
            }
        }

        List<String> answer = new ArrayList<>(checkMap);
        Collections.sort(answer, Collections.reverseOrder());
        for(String p : answer) {
            System.out.println(p);
        }
    }
}